prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -413,36 +408,72 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
      * @param httpClient the HTTP client to make the call
      */
     suspend fun getAuthTokenWithUserPass(
-        restUrl: String,
+        restTransportInfo: RESTTransportType,
         credential: UserPassCredential,
         logger: Logger,
         httpClient: HttpClient,
     ): TokenInfo {
         httpClient.use { client ->
-            if (restUrl.contains("dataingestion.test.nbspreview.com")) {
-                val idTokenInfoString: String = client.post(restUrl) {
-                    val credentialString = credential.user + ":" + credential.pass
-                    val basicAuth = "Basic " + Base64.getEncoder().encodeToString(credentialString.encodeToByteArray())
-                    expectSuccess = true // throw an exception if not successful
-                    postHeaders(
-                        mapOf(
-                            "Authorization" to basicAuth,
-                            "Host" to "dataingestion.test.nbspreview.com"
-                        )
-                    )
-                }.body()
-                logger.info("Got Token with UserPass")
-                return TokenInfo(accessToken = "Bearer " + idTokenInfoString, expiresIn = 3600)
-            } else {
-                val idTokenInfoString: String = client.post(restUrl) {
-                    expectSuccess = true // throw an exception if not successful
-                    contentType(ContentType.Application.Json)
+            val restUrl = restTransportInfo.authTokenUrl
+            val idTokenInfoString: String = client.post(restUrl) {
+                expectSuccess = true // throw an exception if not successful
+                postHeaders(
+                    if (restTransportInfo.authHeaders["Authorization-Type"] == "Basic Auth") {
+                        // Authorization-Type: "Basic Auth" requires the following:
+                        // Header:
+                        //  Authorization : "Basic + base64(username+password)"
+                        // Body:
+                        //  None
+                        // TODO: Add Authorization-Type: "Basic Auth" to DATAINGESTION (NBS) RESTTransport.authHeaders
```

Comment:
I noticed that "Basic Auth" is in the NBS Staging settings.... does this need to be done for prod, too, or is this referring to something different?